kernel/environ.ml

```diff
@@ -568,11 +568,24 @@ let is_primitive env c =
   | Declarations.Primitive _ -> true
   | _ -> false
 
+let is_int63_type env c =
+  match env.retroknowledge.Retroknowledge.retro_int63 with
+  | None -> false
+  | Some c' -> Constant.CanOrd.equal c c'
+
+let is_float64_type env c =
+  match env.retroknowledge.Retroknowledge.retro_float64 with
+  | None -> false
+  | Some c' -> Constant.CanOrd.equal c c'
+
 let is_array_type env c =
   match env.retroknowledge.Retroknowledge.retro_array with
   | None -> false
   | Some c' -> Constant.CanOrd.equal c c'
 
+let is_primitive_type env c =
+  is_int63_type env c || is_float64_type env c || is_array_type env c
```

Comment:
This may be difficult to keep up to date, perhaps we could do
```suggestion
  (* dummy match to force an update if we add a primitive type *)
  let _ = CPrimitives.(function PTE(PT_int63|PT_float64|PT_array) -> ()) in
  is_int63_type env c || is_float64_type env c || is_array_type env c
```